### PR TITLE
Add Min and Max order subtotal threshold for shipper and remove total…

### PR DESCRIPTION
**Pull request by handoyo**

### CODE REVIEW

**Comment by BasaraABC** [comment]
Thank you. Could you please add some description

**Comment by handoyo** [comment]
> Thank you. Could you please add some description

I add min and max order subtotal threshold for shipper based on https://forum.abantecart.com/index.php/topic,10123.msg39250.html#msg39250. 

The poster would like to hide free delivery and show the flat rate if the order is under 75, and when over 75 it shows just free delivery and hide the flat rate.

The min and max fields are optional so admins can leave them empty if the don't want to set the threshold.

And i remove the total from the free shipping as it will bring confusion for admins due to the additional min and max threshold in the settings page.

File: public_html/storefront/model/checkout/extension.php [hunk 1]
```diff
@@ -76,6 +79,18 @@ public function getExtensions($type)
...
                     }
+
+                    //filter by order subtotal limits
+                    $minOrderSubtotal = $this->config->get($row['key']."_min_order_subtotal") ? : 0;
+                    $maxOrderSubtotal = $this->config->get($row['key']."_max_order_subtotal") ? : 0;
+                    $minOrderSubtotal = $this->currency->convert($minOrderSubtotal, $this->config->get('config_currency'), $this->currency->getCode());
+                    $maxOrderSubtotal = $this->currency->convert($maxOrderSubtotal, $this->config->get('config_currency'), $this->currency->getCode());
+
+                    if ($cartSubtotal
+                        && ($cartSubtotal <= $minOrderSubtotal || ($maxOrderSubtotal && $cartSubtotal >= $maxOrderSubtotal))
```
Comment (abolabo) [inline public_html/storefront/model/checkout/extension.php:90]: Why <= and >= operators here? If cart subtotal equal min or max it's a valid value for shipment

Comment (handoyo) [reply public_html/storefront/model/checkout/extension.php:90]: my consideration here is  e.g the min is set to 10, and the order subtotal is 10.00, the shipper is shown. same for the max.

Comment (abolabo) [reply public_html/storefront/model/checkout/extension.php:90]: look at your code. Line 92. 
By logic if order subtotal 10 and min 10 you continue then (shipping not shown). It's a bug

**Comment by abolabo** [comment]
thank you

**Pull request merged**